src/reverse_route.coffee

```diff
@@ -1,11 +1,22 @@
-class ReverseRoute extends String
+class ReverseRoute
 
   constructor: ({@path, @method}) ->
-    super @path
+    @length = @path.length
 
+    # Enable single character access
+    @[i] = character for character, i in @path
 
-  toString: ->
-    @path
+
+  # Forward String methods
+  indexOf: -> @path.indexOf arguments...
+
```

Comment:
Either remove all the empty lines here to make this one code block for forwarding string methods, or if you want to keep these methods separate then format them as real methods, i.e. with the body on a separate line, and a comment. I.e. either:

```coffeescript
# Forward String methods
indexOf: -> @path.indexOf arguments...
match: -> @path.match arguments...
slice: -> @path.slice arguments...
toString: -> @path
valueOf: -> @path
```

or 

```coffeescript
# Forwarded String method
indexOf: -> 
  @path.indexOf arguments...

# Forwarded String method
match: -> 
  @path.match arguments...

# Forwarded String method
slice: -> 
  @path.slice arguments...
```